### PR TITLE
layers: cv: get rid of device-scoped current renderpass tracking

### PR DESCRIPTION
**Pull request by chrisforbes**
This was completely redundant to the (correct) CB-scoped renderpass
tracking, and only served to break things when different threads
simultaneously recorded command buffers.

Signed-off-by: Chris Forbes <chrisforbes@google.com>

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -9477,7 +9469,6 @@ VK_LAYER_EXPORT VKAPI_ATTR void VKAPI_CALL vkCmdNextSubpass(VkCommandBuffer comm
     layer_data *dev_data = get_my_data_ptr(get_dispatch_key(commandBuffer), layer_data_map);
     loader_platform_thread_lock_mutex(&globalLock);
     GLOBAL_CB_NODE *pCB = getCBNode(dev_data, commandBuffer);
-    TransitionSubpassLayouts(commandBuffer, &dev_data->renderPassBeginInfo, ++dev_data->currentSubpass);
```
Comment (lentinem) [inline layers/core_validation.cpp]: It's hard to tell from the diff but I assume this is still called elsewhere?

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -9529,7 +9520,6 @@ VK_LAYER_EXPORT VKAPI_ATTR void VKAPI_CALL vkCmdEndRenderPass(VkCommandBuffer co
     }
 #endif
     GLOBAL_CB_NODE *pCB = getCBNode(dev_data, commandBuffer);
-    TransitionFinalSubpassLayouts(commandBuffer, &dev_data->renderPassBeginInfo);
```
Comment (lentinem) [inline layers/core_validation.cpp]: Same here.

File: layers/core_validation.cpp (hunk 1 above)
Comment (chrisforbes) [inline layers/core_validation.cpp]: Yes -- just a few lines away, with the correct per-CB state.

**Comment by lentinem** [comment]
+1

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**